Push/PushCrypto.swift

```diff
@@ -1,382 +0,0 @@
-// This Source Code Form is subject to the terms of the Mozilla Public
-// License, v. 2.0. If a copy of the MPL was not distributed with this
-// file, You can obtain one at http://mozilla.org/MPL/2.0/
-
-import Foundation
-import Common
-
-/// Class to wrap ecec which does the encryption, decryption and key generation with OpenSSL.
-/// This supports aesgcm and the newer aes128gcm.
-/// For each standard of decryption, two methods are supplied: one with Data parameters and return value,
-/// and one with a String based one.
-class PushCrypto {
-    // Stateless, we provide a singleton for convenience.
-    public static var sharedInstance = PushCrypto()
-    private let logger: Logger
-
-    init(logger: DefaultLogger = DefaultLogger.shared) {
-        self.logger = logger
-    }
-}
-
-// AES128GCM
-extension PushCrypto {
-    func aes128gcm(payload data: String, decryptWith privateKey: String, authenticateWith authKey: String) throws -> String {
-        guard let authSecret = authKey.base64urlSafeDecodedData,
-            let rawRecvPrivKey = privateKey.base64urlSafeDecodedData,
-            let payload = data.base64urlSafeDecodedData else {
-                throw PushCryptoError.base64DecodeError
-        }
-
-        let decrypted = try aes128gcm(payload: payload,
-                                      decryptWith: rawRecvPrivKey,
-                                      authenticateWith: authSecret)
-
-        guard let plaintext = decrypted.utf8EncodedString else {
-            throw PushCryptoError.utf8EncodingError
-        }
-
-        return plaintext
-    }
-
-    func aes128gcm(payload: Data, decryptWith rawRecvPrivKey: Data, authenticateWith authSecret: Data) throws -> Data {
-        var plaintextLen = ece_aes128gcm_plaintext_max_length(payload.getBytes(), payload.count) + 1
```

Comment:
This isn't used anywhere else in the project, I wonder if we could remove it as well. Location [here](https://github.com/mozilla-mobile/firefox-ios/blob/cafa48130344b0f60ce33e852e4b303d8647c408/ThirdParty/ecec/src/decrypt.c#L325). There seems to be other clean up we could do under this ThirdParty/ecec/src folder

Reply:
Yes! I'm getting close to cleaning all the low-level crypto code soon in https://github.com/mozilla-mobile/firefox-ios/issues/16004! Which will remove **all the openssl bindings**  including this one